src/app.py

```diff
@@ -2566,10 +2571,9 @@ def update_uploads_datastatus():
                 if isinstance(upload[prop], str) and \
                         len(upload[prop]) >= 2 and \
                         upload[prop][0] == "[" and upload[prop][-1] == "]":
-                    upload[prop] = upload[prop].replace("'", '"')
-                    upload[prop] = json.loads(upload[prop])
-                    if len(upload[prop]) > 0:
-                        upload[prop] = upload[prop][0]
+                    prop_as_list = string_helper.convert_str_literal(upload[prop])
+                    if len(prop_as_list) > 0:
```

Comment:
Same as above:
- check the data type instance after string evaluation and handle string/list accordingly
- change `upload[prop] = " "` to an empty string or empty `[]` based on the type

Reply:
Same as above